src/spicy/spicyz/glue-compiler.cc

```diff
@@ -38,12 +38,8 @@ static std::string::size_type looking_at(const std::string& chunk, std::string::
                                          const std::string_view& token) {
     eat_spaces(chunk, &i);
 
-    for ( char j : token ) {
-        if ( i >= chunk.size() || chunk[i++] != j )
-            return 0;
-    }
-
-    return i;
+    bool token_at_position = token == std::string_view(chunk).substr(i, token.size());
```

Comment:
I think there is a behavior change here if `i > chunk.size()` it would now throw an exception I think

Maybe add an assertion? I don't think it'd hit that case in real execution though

Reply:
Uff, good catch. I added a check that `i` does not exceed the chunk size `i < chunk.size()`. That makes the code slightly dense, but reproduces the previous behavior.